bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -204,12 +210,16 @@ ClientsManager::ClientsManager(const std::set<NodeIdType>& proxyClients,
       clientServices_{clientServices},
       internalClients_{internalClients},
       maxReplySize_(ReplicaConfig::instance().getmaxReplyMessageSize()),
-      maxNumOfReqsPerClient_(
-          ReplicaConfig::instance().clientBatchingEnabled ? ReplicaConfig::instance().clientBatchingMaxMsgsNbr : 1),
+      maxNumOfReqsPerClient_(ReplicaConfig::instance().clientBatchingEnabled &&
+                                     ReplicaConfig::instance().preExecutionFeatureEnabled
```

Comment:
Why to ask about ReplicaConfig::instance().preExecutionFeatureEnabled?

Reply:
As far as I see, client batch is possible only when pre-processing is enabled. I added this to reduce the number of reserved pages when it's possible.

Reply:
Exactly, that's why there is no need to ask if preExecutionFeatureEnabled or not.

Reply:
Theoretically it's possible to set in configurations _clientBatchingEnabled=True_ along with _preExecutionFeatureEnabled=False_.
I agree that in this case the _clientBatchingEnabled=True_ doesn't have a reasonable meaning but I don't see that we force anywhere clientBatchingEnabled to be False when pre-executing is disable. 
So this "extra" checking is for that not-smart-but-possible case.